presto-main/src/main/java/com/facebook/presto/operator/HashBuilderOperator.java

```diff
@@ -524,6 +525,13 @@ private void finishInput()
             return;
         }
 
+        checkState(index != null, "index is null");
+        ListenableFuture<?> reserved = localUserMemoryContext.setBytes(index.getEstimatedMemoryRequiredToCreateLookupSource(
+                sortChannel));
```

Comment:
Noob question, from the description of `localUserMemoryContext.setBytes`, "When this method returns, the bytes tracked by this LocalMemoryContext has been updated." If we call the finishInput multiple times due to previous calls have "if (!reserved.isDone())", will the memory request be counted multiple times?

Reply:
My understanding is that `setBytes` work like a gauge, so you just set the whatever watermark at the time and it would automatically update the difference to the upstream: https://github.com/prestodb/presto/blob/master/presto-memory-context/src/main/java/com/facebook/presto/memory/context/SimpleLocalMemoryContext.java#L82
Therefore calling finishInput multiple times shouldn't result into memory request being counted multiple times.